File: 算法基础课(修订版)/循环/P14073/verify_acm.py
class Solution:
    def solve(self, left_weights, right_weights):
        left_total = sum(left_weights)
        right_total = sum(right_weights)
        return "Equal" if left_total == right_total else "Not Equal"


if __name__ == "__main__":
    n, m = map(int, input().split())
    left_weights = list(map(int, input().split()))
    right_weights = list(map(int, input().split()))

    solution = Solution()
    result = solution.solve(left_weights, right_weights)

    print(result)

File: 算法基础课(修订版)/循环/P14073/verify_acm.cc
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    string solve(vector<int>& leftWeights, vector<int>& rightWeights) {
        int leftTotal = 0;
        int rightTotal = 0;
        for (int w : leftWeights) leftTotal += w;
        for (int w : rightWeights) rightTotal += w;
        return leftTotal == rightTotal ? "Equal" : "Not Equal";
    }
};

int main() {
    int n, m;
    cin >> n >> m;

    vector<int> leftWeights(n);
    vector<int> rightWeights(m);

    for (int i = 0; i < n; ++i) {
        cin >> leftWeights[i];
    }
    for (int i = 0; i < m; ++i) {
        cin >> rightWeights[i];
    }

    Solution solution;
    cout << solution.solve(leftWeights, rightWeights) << endl;

    return 0;
}
